clippy_lints/src/enum_glob_use.rs

```diff
@@ -25,17 +25,7 @@ declare_clippy_lint! {
     "use items that import all variants of an enum"
 }
 
-pub struct EnumGlobUse;
-
-impl LintPass for EnumGlobUse {
-    fn get_lints(&self) -> LintArray {
-        lint_array!(ENUM_GLOB_USE)
-    }
-
-    fn name(&self) -> &'static str {
-        "EnumGlobUse"
-    }
-}
+declare_lint_pass!(EnumGlobUse => [ENUM_GLOB_USE]);
```

Comment:
I'll review this in batches. Until here, everything looks good 👍 

Also -1505 net LoC 🎉

Reply:
I had to change the name returned for `entry::HashMapPass` and `minmax::MinMaxPass` to avoid errors or lints.  I hope that's OK.

Reply:
Please see my subsequent comments for the errors that I encountered.